R/PipeOpPredictionUnion.R

```diff
@@ -0,0 +1,136 @@
+#' * `param_vals` :: named `list`\cr
+#'   List of hyperparameter settings, overwriting the hyperparameter settings that would otherwise
+#'   be set during construction. Default `list()`.
+#'
+#' @section Input and Output Channels:
+#' [`PipeOpPredictionUnion`] has multiple input channels depending on the `innum` construction
+#' argument, named `"input1"`, `"input2"`, ... if `innum` is nonzero; if `innum` is 0, there is only
+#' one *vararg* input channel named `"..."`. All input channels take `NULL` during training and a
+#' [`Prediction`][mlr3::Prediction] during prediction.
+#'
+#' [`PipeOpPredictionUnion`] has one output channel named `"output"`, producing `NULL` during
+#' training and a [`Prediction`][mlr3::Prediction] during prediction.
+#'
+#' The output during prediction is a [`Prediction`][mlr3::Prediction] constructed by combining all
+#' input [`Prediction`][mlr3::Prediction]s.
+#'
+#' @section State:
+#' The `$state` is left empty (`list()`).
+#'
+#' @section Parameters:
+#' [`PipeOpPredictionUnion`] has no Parameters.
+#'
+#' @section Internals:
+#' Only sets the fields `row_ids`, `truth`, `response` and if applicable `prob` and `se` during
+#' construction of the output [`Prediction`][mlr3::Prediction].
+#'
+#' @section Fields:
+#' Only fields inherited from [`PipeOp`].
+#'
+#' @section Methods:
+#' Only methods inherited from [`PipeOp`].
+#'
+#' @family PipeOps
+#' @include PipeOp.R
+#' @export
+#' @examples
+#' library("mlr3")
+#'
+#' task = tsk("iris")
+#' filter = expression(Sepal.Length < median(Sepal.Length))
+#' gr = po("copy", outnum = 2) %>>% gunion(list(
+#'   po("filterrows", id = "filter1",
+#'     param_vals = list(filter = filter)) %>>%
+#'   lrn("classif.rpart", id = "learner1"),
+#'   po("filterrows", id = "filter2",
+#'      param_vals = list(filter = filter, invert = TRUE)) %>>%
+#'   lrn("classif.rpart", id = "learner2")
+#' )) %>>% po("predictionunion")
+#'
+#' gr$train(task)
+#' gr$predict(task)
+PipeOpPredictionUnion = R6Class("PipeOpPredictionUnion",
+  inherit = PipeOp,
+  public = list(
+    initialize = function(innum = 0L, id = "predictionunion", param_vals = list()) {
+      assert(
+        check_int(innum, lower = 0L),
+        check_character(innum, min.len = 1L, any.missing = FALSE)
+      )
+      if (!is.numeric(innum)) {
+        innum = length(innum)
+      }
+      inname = if (innum) rep_suffix("input", innum) else "..."
+      super$initialize(id, param_vals = param_vals,
+        input = data.table(name = inname, train = "NULL", predict = "Prediction"),
+        output = data.table(name = "output", train = "NULL", predict = "Prediction"))
+    }
+  ),
+  private = list(
+    .train = function(inputs) {
+      self$state = list()
+      list(NULL)
+    },
+    .predict = function(inputs) {
+      # currently only works for task_type "classif" or "regr"
+      check = all((unlist(map(inputs[-1L], .f = `[[`, "task_type")) == inputs[[1L]]$task_type) &
```

Comment:
when you know the supposed result of the `map()` call, you can use the `map_xxx` functions, which are basically like `vapply()` only that you have to remember more names. In this case `map_chr()`.

Reply:
also with `map()` and its cousins you can just do `map_chr(inputs[-1], "task_type")`